server/app/filters/DisableCachingFilter.java

```diff
@@ -22,11 +29,29 @@ public EssentialAction apply(EssentialAction next) {
         request ->
             next.apply(request)
                 .map(
-                    result ->
-                        result
-                            .withHeader("Cache-Control", "no-cache, must-revalidate")
-                            .withHeader("Pragma", "no-cache")
-                            .withHeader("Expires", "0"),
+                    result -> {
+                      String path = request.uri();
+                      Integer status = result.status();
+                      ImmutableList<String> assets = staticAssetPaths;
+
+                      if (environment.isDev()) {
+                        // Must revalidate status asset caches in dev mode
+                        assets = ImmutableList.<String>of();
+                      }
+                      if (assets.stream().anyMatch(m -> path.contains(m))) {
```

Comment:
Consider using `startsWith` rather than `contains`.

I think this would be more accurate (e.g. not catch resources like `some-dynamic-admin-path/public/...`) and more performant (not have to scan the whole string to return false). The performance concern is unlikely to matter, but the correctness concern might.

Reply:
Agree that doing startsWith is preferrable behavior for correctness. Could do
```java
private static final Splitter PATH_SPLITTER = Splitter.on("/');
...
Optional<String> pathPrefix = Iterables.getFirst(PATH_SPLITTER.splitToList(path));
if (pathPrefix.map(ASSET_PATH_PREFIXES::contains).orElse(false)) {
```
with that approach could also change the hard-coded list of path prefixes to an `ImmutableSet`

Reply:
done